### PR TITLE
Add interface PaymentRequest into WebIDL

### PR DESCRIPTION
**Pull request by bentian**
Add following files:
1. dom/webidl/PaymentRequest.webidl
2. dom/payments/PaymentRequest.{h, cpp}

Notes:
- PaymentRequest.{h, cpp} are based on files generated from ./mach webidl-example. Prototype only.
- **Make PaymentDetails.total _required_** to avoid WebIDL error in #5.
- Build pass only, require further verification.

### CODE REVIEW

File: dom/webidl/PaymentRequest.webidl [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * The origin of this WebIDL file is
+ *   https://www.w3.org/TR/payment-request/#paymentrequest-interface
+ */
+
+dictionary PaymentMethodData {
+  required sequence<DOMString> supportedMethods;
+           object              data;
```
Comment (marcoscaceres) [inline dom/webidl/PaymentRequest.webidl]: Nit: is it usual for us to indent things this way in IDL files?

File: dom/webidl/PaymentRequest.webidl [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  Promise<PaymentResponse> show();
+  Promise<void>            abort();
+  Promise<boolean>         canMakePayment();
+
+  readonly attribute DOMString?           paymentRequestID;
+  readonly attribute PaymentAddress?      shippingAddress;
+  readonly attribute DOMString?           shippingOption;
+  readonly attribute PaymentShippingType? shippingType;
+
+  // Supports "shippingaddresschange" event
```
Comment (marcoscaceres) [inline dom/webidl/PaymentRequest.webidl]: nit: I know this is in the spec, but that comment doesn't make any sense... I'll remove it from the spec too :)

File: dom/webidl/PaymentRequest.webidl (hunk 1 above)
Comment (bentian) [inline dom/webidl/PaymentRequest.webidl]: The indention follows [PaymentMethodData on spec](https://www.w3.org/TR/payment-request/#paymentmethoddata-dictionary). I copied and pasted it here. I'm fine if you prefer to remove.

File: dom/webidl/PaymentRequest.webidl (hunk 2 above)
Comment (bentian) [inline dom/webidl/PaymentRequest.webidl]: OK I'll remove the comment.

File: dom/webidl/PaymentRequest.webidl (hunk 1 above)
Comment (bentian) [inline dom/webidl/PaymentRequest.webidl]: Both styles appear under dom/webidl, e.g., [Touch.webidl](searchfox.org/mozilla-central/source/dom/webidl/Touch.webidl) contains both. I'm fine for either one.

Comment (bentian) [inline dom/webidl/PaymentRequest.webidl]: I'll keep the indention for now to follow spec and make it clear. Feel free to revise indention style with another PR.

Comment (marcoscaceres) [inline dom/webidl/PaymentRequest.webidl]: ok, np

**Comment by marcoscaceres** [comment]
Merge if happy 👍

**Pull request merged**